test/unittest/tests/block/test_block_session.cpp

```diff
@@ -0,0 +1,286 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * [block_session]: block_session.c
+ * The block manager extent list consists of both extent and size type blocks. This unit test
+ * suite tests aims to test all of the allocation and frees of the extent and size block functions.
+ *
+ * The block session manages an internal caching mechanism for both block and size blocks that are
+ * created or discarded.
+ */
+#include "wt_internal.h"
+#include <catch2/catch.hpp>
+#include "../wrappers/mock_session.h"
+
+void
+validate_ext_block(WT_EXT *ext)
+{
+    REQUIRE(ext != nullptr);
+    REQUIRE(ext->depth != 0);
+    REQUIRE(ext->size == 0);
+    REQUIRE(ext->off == 0);
+
+    for (int i = 0; i < ext->depth; i++) {
+        REQUIRE(ext->next[i + ext->depth] == nullptr);
+    }
+}
+
+void
+free_ext_block(WT_EXT *ext)
+{
+    __wt_free(nullptr, ext);
+}
+
+void
+validate_ext_list(WT_BLOCK_MGR_SESSION *bms, int expected_items)
+{
+    REQUIRE(bms != nullptr);
+    WT_EXT *curr = bms->ext_cache;
```

Comment:
As this code is C++, and our coding standard allows it, variables can be declared for the smaller scope. I suggest moving the declaration of `curr` to when it is first required - ie immediately before the `for` loop.

Reply:
Thanks fixed. [5f71e2e](https://github.com/wiredtiger/wiredtiger/pull/10963/commits/5f71e2efe324cb7f0f225d03d07a9041839c75f4)

Reply:
That was quick :-)